### PR TITLE
pagination works

### PR DESCRIPTION
**Pull request by choiii-darren**
#53

### CODE REVIEW

File: app/routes.py [hunk 1]
```diff
@@ -110,26 +113,43 @@ def categoryHandler(id):
...
+    matches = Matches.query.all()
+    matches = matches[::-1]
+    matchPage = matches[(int(page)-1)*10:int(page)*10]
+    return json.dumps([m.serialize() for m in matchPage])
+
+
+@app.route('/api/matches/count', methods=['GET'])
+def getMatchesCount():
+    matches = Matches.query.all()
+    return str(int(len(matches)))
```
Comment (ivanccheng) [inline app/routes.py:148]: Looks good! 

As far as I can see, the getMatchesCount gets all of the matches, which is already happening in getMatchPage call. Is there a way that we can get the length in the getMatchPage query? 

This implementation works for now, but what happens if we want to change the length of results per page as a request in the future or we want to enhance it to also query based on date? Could we use REST API parameters to pass this information?

File: app/routes.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+from audioop import cross
```
Comment (ivanccheng) [inline app/routes.py:1]: just vscode things

**Review by ivanccheng — COMMENTED** [review]
Nice work!

some minor ui edge cases that I saw when playing around with it:
- when clicking final page in mobile, the pagination bar loses the right padding 
- when clicking right array alot in mobile, same issue as above
image
image

nit(pick):
changing the page doesn't wait for results to reload and refresh

File: app/routes.py (hunk 1 above)
Comment (choiii-darren) [reply app/routes.py:148]: ill take a loop at the resposniveness, and ig i can return metadata about the matches? but that would have to be in the header....
regarding the length of results, that's just a magic number that i can also turn into a variable for easy change. idk how to do REST API parameters tho

Comment (ivanccheng) [reply app/routes.py:148]: Something that can work is the make /api/matches return a json object like this:
```
{ 
    metadata: {
        page: int
        max_page: int
        number_matches: int
        other_useful_data: 
    }
    records: [
        match1,
        match2, 
        etc
    ]
}
```
For parameters, in flask you can access the URL parameters by using `request.args` 
https://flask.palletsprojects.com/en/2.2.x/api/#flask.Request.args

So in the end, an example request might look like `base_url/api/matches?page=2&results_per_page=20` or something,
Example here: https://www.botify.com/learn/basics/what-are-url-parameters

File: app/routes.py [hunk 3]
```diff
@@ -110,26 +113,48 @@ def categoryHandler(id):
...
+    # app.logger.debug('hit match pagination')
+    page = request.args.get('page')
+    matchPerPage = 10
+    matches = Matches.query.all()
+    matches = matches[::-1]
+    matchPage = matches[(int(page)-1)*matchPerPage:int(page)*matchPerPage]
+    returnObj = {
+        'metadata':
+        {
+            'matchCount': str(int(len(matches))),
```
Comment (ivanccheng) [inline app/routes.py:147]: lots of type casting here, is there a reason why we we need to cast `len(matches)` to an int?

File: app/routes.py [hunk 4]
```diff
@@ -110,26 +113,48 @@ def categoryHandler(id):
...
-def getMatches():
-    app.logger.debug('This endpoint was hit!')
-    matches = Matches.query.all()
-    return json.dumps([m.serialize() for m in matches])
 
-@cross_origin()
-@app.route("/api/match/<id>", methods = ["DELETE", "GET", "PUT"])
+
+# @cross_origin()
+# @app.route("/api/matches", methods=["GET"])
```
Comment (ivanccheng) [inline app/routes.py:128]: just realized that we still might need this for the admin page, and for andrew, maybe we can move the old get all matches to something like api/matches/all for now, and later we can all the all functionality using request params

**Review by ivanccheng — COMMENTED** [review]
Minor changes, before we merge, lets deploy it and play around with it, and maybe get someone who’s got a keen eye for design to take a look too! Good work using query params

FYI: I added some nitpicks and labeled them as such, these don’t necessarily need to change but I felt they could be good, and provided some reasoning for it, it’s up to you to implement or not.

File: app/routes.py [hunk 5]
```diff
@@ -110,26 +112,51 @@ def categoryHandler(id):
...
+        matchPerPage = 10
+        matches = Matches.query.all()
+        matches = matches[::-1]
+        matchPage = matches[(int(page)-1)*matchPerPage:int(page)*matchPerPage]
+        returnObj = {
+            'metadata':
+            {
+                'recordCount': str(len(matches)),
+                'pageCount': str(int(len(matches)/matchPerPage) + 1),
+                'recordsPerPage': str(matchPerPage),
```
Comment (ivanccheng) [inline app/routes.py:149]: nit: how come you need to do all of this type casting?

**Review by ivanccheng — APPROVED** [review]
good works, nit picks only

File: app/routes.py (hunk 5 above)
Comment (choiii-darren) [reply app/routes.py:149]: something weird happened with returning obj but seemed to work with strings so I left it as

File: app/routes.py [hunk 6]
```diff
@@ -110,26 +112,51 @@ def categoryHandler(id):
...
+@cross_origin
+@app.route('/api/matches', methods=['GET'])
+def getMatchPage():
+    try:
+        # app.logger.debug('hit match pagination')
+        page = request.args.get('page')
+        matchPerPage = 10
+        matches = Matches.query.all()
+        matches = matches[::-1]
+        matchPage = matches[(int(page)-1)*matchPerPage:int(page)*matchPerPage]
```
[the remark this answers is not in the document]
Comment (choiii-darren) [reply app/routes.py:143]: will be fixed with the default query branch

**Pull request merged**